src/brevitas/graph/gpfq.py

```diff
@@ -280,103 +125,124 @@ def update_batch(self, module, input, current_layer):
             current_layer.forward_count = 0
             raise StopFwdException
 
-    def _get_permutation_list(self, weight: Tensor):
-        permutation_list = []
-        if self.act_order:
-            # We don't need full Hessian, we just need the diagonal
-            H_diag = self.quant_input.transpose(2, 1).square().sum(2)
-            for group_index in range(self.groups):
-                # Re-order Hessian_diagonal so that weights associated to
-                # higher magnitude activations are quantized first
-                perm = torch.argsort(H_diag[group_index, :], descending=True)
-                perm = perm.to(weight.device)
-                permutation_list.append(perm)
-        else:
-            for group_index in range(self.groups):
-                # No permutation, permutation tensor is a ordered index
-                perm = torch.tensor(range(weight.shape[-1]), device=weight.device)
-                permutation_list.append(perm)
-        return permutation_list
+    def get_quant_weights(self, g, t, permutation_list) -> Tensor:
+        # TODO: we would like to implement slicing to minimize the amount
```

Comment:
I can't think of a reason why slicing can't be done for a slice rather than a single value. We might need to change the shared function signature a bit, but I'd rather do that than having a completely new function